nexus-tweaks.user.js

```diff
@@ -32,6 +32,391 @@ function timesCharExist(x, c){
+      if (statusMatch[2].indexOf("HealthBar") < 0
+          && statusMatch[2].indexOf("MagicBar") < 0
+          && statusMatch[1].indexOf("Sorcerers Might") < 0) {
+        var newStatus = {
+          'abbrev': statusMatch[1].substring(0, 2),
+          'title': statusMatch[1],
+          'class': 'status-tag-standard'
+        };
+        checkStatus(newStatus, 'Evil', 'E', 'status-tag-ev');
+        checkStatus(newStatus, 'Neutral', 'N', 'status-tag-nt');
+        checkStatus(newStatus, 'Good', 'G', 'status-tag-gd');
+        checkStatus(newStatus, 'Hidden', 'H', 'status-tag-positive');
+        checkStatus(newStatus, 'Flying', 'F', 'status-tag-positive');
+        checkStatus(newStatus, 'Hellfire', 'HF', 'status-tag-negative');
+        checkStatus(newStatus, 'Defiler Poison', 'DP', 'status-tag-negative');
+        checkStatus(newStatus, 'cursed by', 'AC', 'status-tag-negative');
+        person.statusTags.push(newStatus);
+      }
+      // remove processed status, move on to next one
+      charHtml = charHtml.replace(statusMatch[0], '');
+      statusMatch = statusRegExp.exec(charHtml);
+    }
+    return person;
+  }
+
+  const characterListTweaks = async (mod) => {
+    var coordArea = document.getElementById('AreaDescription');
+    if (!coordArea) {
+      // dead
+      localStorage.clear();
+      return;
+    }
+    var charArea = document.getElementsByClassName('charListArea')[0];
+    var peopleMatch = /There (is|are) (\d+) other (person|people) here, (.*)/
+    .exec(charArea.innerHTML);
+
+    var location = /(\d+, \d+ .+?),/.exec(coordArea.innerHTML)[1];
+    var previousLocation = localStorage.getItem('last-location');
+    var locationUnchanged = (location === previousLocation);
+    localStorage.setItem('last-location', location);
+
+    if (!peopleMatch || (peopleMatch[2] === '0')) {
+      // no people present, can clear saved previous HP values
+      localStorage.clear();
+      localStorage.setItem('last-location', location);
+      return;
+    }
+    var charHtmls = peopleMatch[4].substring(1, peopleMatch[4].length - 1).split(
+        '>, <');
+    var charCount = charHtmls.length;
+    var personList = [];
+
+    // create list of people and calculate sorting value
+    for (let i = 0; i < charCount; i++) {
+      var element = charHtmls[i];
+      var newPerson = createPerson(element);
+      newPerson.sortValue = calculateSortValue(newPerson);
+      personList.push(newPerson);
+    }
+
+    // sort list of people, higher sortValue sorts to top of list
+    personList = personList.sort((a, b) => b.sortValue - a.sortValue);
+
+    // save list of people previously seen on last refresh.
+    // used to remove entries for people no longer present later
+    var peopleToDelete = Object.keys(localStorage).filter(key => key !== 'last-location');
+    var peopleJoined = [];
+
+    // begin create html
+    var resultHtml = "";
+    for (let i = 0; i < personList.length; i++) {
```

Comment:
comment: The cognitive complexity of this method is getting high. See if you can break out any portions of this function into other functions.

Specifically, starting with any logic you're placing into for loops is a good place to look at refactoring.

Reply:
(WIP) will check this out

Reply:
refactored implementation is in new commit (I will squash it, but so you can check that out more easily)